server/src/main/java/org/elasticsearch/index/mapper/TimestampFieldMapper.java

```diff
@@ -0,0 +1,255 @@
+            return new TimestampFieldType();
+        }
+
+        @Override
+        public String typeName() {
+            return NAME;
+        }
+
+        @Override
+        public Query termQuery(Object value, QueryShardContext context) {
+            throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() + "] does not support term queries");
+        }
+
+        @Override
+        public Query existsQuery(QueryShardContext context) {
+            throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() + "] does not support exists queries");
+        }
+
+    }
+
+    public static class Builder extends MetadataFieldMapper.Builder<Builder> {
+
+        private String path;
+
+        public Builder() {
+            super(NAME, Defaults.TIMESTAMP_FIELD_TYPE);
+        }
+
+        public void setPath(String path) {
+            this.path = path;
+        }
+
+        @Override
+        public MetadataFieldMapper build(BuilderContext context) {
+            return new TimestampFieldMapper(
+                fieldType,
+                new TimestampFieldType(),
+                path
+            );
+        }
+    }
+
+    public static class TypeParser implements MetadataFieldMapper.TypeParser {
+
+        @Override
+        public MetadataFieldMapper.Builder<?> parse(String name,
+                                                    Map<String, Object> node,
+                                                    ParserContext parserContext) throws MapperParsingException {
+            Builder builder = new Builder();
+            for (Iterator<Map.Entry<String, Object>> iterator = node.entrySet().iterator(); iterator.hasNext();) {
+                Map.Entry<String, Object> entry = iterator.next();
+                String fieldName = entry.getKey();
+                Object fieldNode = entry.getValue();
+                if (fieldName.equals("path")) {
+                    builder.setPath((String) fieldNode);
+                    iterator.remove();
+                }
+            }
+            return builder;
+        }
+
+        @Override
+        public MetadataFieldMapper getDefault(ParserContext parserContext) {
+            return new TimestampFieldMapper(Defaults.TIMESTAMP_FIELD_TYPE,
+                new TimestampFieldType(), null);
+        }
+    }
+
+    private final String path;
+
+    private TimestampFieldMapper(FieldType fieldType, MappedFieldType mappedFieldType, String path) {
+        super(fieldType, mappedFieldType);
+        this.path = path;
+    }
+
+    public void validate(DocumentFieldMappers lookup) {
+        if (path == null) {
+            // not configured, so skip the validation
+            return;
+        }
+
+        Mapper mapper = lookup.getMapper(path);
+        if (mapper == null) {
+            throw new IllegalArgumentException("timestamp meta field's field_name [" + path + "] points to a non existing field");
```

Comment:
I think these messages are intended to be end-user friendly? So something like this could be clearer: "The configured timestamp field does not exist."